src/Tinfoil/Random.hs

```diff
@@ -80,3 +82,10 @@ drawOnce as = do
     n = NE.length as
 
     discard = dropMS extraBits
+
+-- | Generate identifier unique to the currently-running kernel.
+--
+-- https://github.com/torvalds/linux/blob/22763c5cf3690a681551162c15d34d935308c8d7/drivers/char/random.c
+linuxBootId :: IO BootId
+linuxBootId =
+  fmap (BootId . T.encodeUtf8) $ T.readFile "/proc/sys/kernel/random/boot_id"
```

Comment:
Yeah the more I think about it the less I like this idea. Give me a bit and I'll come up with something else.